frontend/packages/ceph-storage-plugin/integration-tests/tests/add-capacity.scenario.ts

```diff
@@ -0,0 +1,108 @@
+  execSync(`kubectl get -o json -n ${NAMESPACE} ${KIND}`).toString(),
+);
+
+let clusterJSON;
+let previousCnt;
+let previousPods;
+let updatedClusterJSON;
+let updatedCnt;
+let updatedPods;
+
+describe('Check add capacity functionality for ocs service', () => {
+  beforeAll(async () => {
+    // check whether the cluster is present or not, and check for its ready state
+    // if not present exit and console that no cluster is present, will be send in next PR
+
+    const { name } = storageCluster.items[0].metadata;
+    clusterJSON = JSON.parse(
+      execSync(`kubectl get -o json -n ${NAMESPACE} ${KIND} ${name}`).toString(),
+    );
+    previousCnt = _.get(clusterJSON, 'spec.storageDeviceSets[0].count', undefined);
+    const uid = _.get(clusterJSON, 'metadata.uid', undefined).toString();
+    previousPods = JSON.parse(execSync(`kubectl get pods -n ${NAMESPACE} -o json`).toString());
+
+    await browser.get(
+      `${appHost}/k8s/ns/openshift-storage/operators.coreos.com~v1alpha1~ClusterServiceVersion`,
+    );
+
+    await click(ocsOp);
+    await click(storageClusterView);
+
+    await browser.wait(until.presenceOf(storageClusterRow(uid)));
+    const kebabMenu = storageClusterRow(uid).$('button[data-test-id="kebab-button"]');
+
+    await click(kebabMenu);
+    await click(addCapacityLbl);
+    await click(addCapacityBtn);
+
+    updatedClusterJSON = JSON.parse(
+      execSync(`kubectl get -o json -n ${NAMESPACE} ${KIND} ${name}`).toString(),
+    );
+
+    updatedCnt = _.get(updatedClusterJSON, 'spec.storageDeviceSets[0].count', undefined);
+    const statusCol = storageClusterRow(uid).$('td:nth-child(4)');
+
+    // need to wait as cluster states fluctuates for some time. Waiting for 2 secs for the same
+    await browser.sleep(2000);
+
+    await browser.wait(until.textToBePresentInElement(statusCol, 'Progressing'));
+    await browser.wait(
+      until.textToBePresentInElement(statusCol.$('span.co-icon-and-text span'), 'Ready'),
+    );
+
+    updatedPods = JSON.parse(execSync(`kubectl get pod -o json -n ${NAMESPACE}`).toString());
+  }, EXPANDWAIT);
+
+  it('Newly added capacity should takes into effect at the storage level', () => {
+    // by default 2Tib capacity is being added
+    expect(updatedCnt - previousCnt).toEqual(1);
+  });
+
+  it('New osd pods corresponding to the additional capacity should be in running state', () => {
+    const isPodPresent = (podName) =>
+      previousPods.items.find((pod) => pod.metadata.name === podName) !== undefined;
+
+    const newPods = [];
+    updatedPods.items.forEach((pod) => {
+      if (!isPodPresent(pod.metadata.name) && pod.metadata.name.startsWith('rook-ceph-osd')) {
+        newPods.push(pod.metadata.name);
+      }
+    });
+
+    /* since rook-ceph-osd-prepare-ocs-deviceset- keeps changing their last 4 characters,
+    hence subtracting the count of previous  rook-ceph-osd-prepare-ocs-deviceset- pods */
+    expect(newPods.length - previousCnt * 3).toEqual(6);
+  });
```

Comment:
The whole block looks flawed to me. Remember that we have `rook-ceph-osd-` pods and `rook-ceph-osd-prepare-` pods. For expansion to be considered complete, new `rook-ceph-osd-` pods must be created. If only `rook-ceph-osd-prepare-` pods are created expansion is not complete and there could be a severe error.

Need more eyes on this.

Reply:
I am checking for both `rook-ceph-osd- ` and `rook-ceph-osd-prepare-`. Thats why the count is being checked with 6 and not 3 for `2Tib` expansion. Can you explain a bit more?

Reply:
I would expect this test to be broken into 2 pieces. 1 for `rook-ceph-osd-` and other for `rook-ceph-osd-prepare-`. But, I'm ok with this for now. Just update the comment to be clearer.